bson/bsoncodec/bsoncodec.go

```diff
@@ -119,13 +119,63 @@ func (vde ValueDecoderError) Error() string {
 // value.
 type EncodeContext struct {
 	*Registry
+
+	// MinSize, if true, instructs encoders to marshal Go integer values (int, int8, int16,
+	// int32, or int64) as the minimum BSON int size (either 32-bit or 64-bit) that can represent
+	// the integer value.
+	//
+	// Deprecated: Use IntMinSize instead.
 	MinSize bool
+
+	// AllowUnexportedFields, if true, instructs encoders to marshal values from unexported struct
+	// fields.
+	AllowUnexportedFields bool
```

Comment:
`AllowUnexportedFields` does not appear tested. Suggest adding tests for all new options on `EncodeContext` and `DecodeContext`.

Reply:
Added tests that assert the expected behavior of the new `Encoder`/`Decoder` configuration methods.